src/cmd/services/m3coordinator/downsample/id_pool_types.go

```diff
@@ -125,23 +125,53 @@ func (p *rollupIDProvider) reset(
 	p.index = -1
 	p.newName = newName
 	p.tagPairs = tagPairs
-	p.nameTagIndex = -1
-	p.rollupTagIndex = -1
-	for idx, pair := range tagPairs {
-		if p.nameTagIndex == -1 && bytes.Compare(p.nameTagBytes, pair.Name) < 0 {
-			p.nameTagIndex = idx
+
+	mergeTags := []struct {
```

Comment:
Can we create this two length array on the provider to avoid allocations? Otherwise potentially this could be heap allocated per call..

Reply:
why would it be heap allocated? it's a local var